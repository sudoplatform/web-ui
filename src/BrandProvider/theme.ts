import { DefaultTheme } from 'styled-components'

const defaultColors = {
  primary: '#3770f6',
  success: '#1ccd82',
  error: '#ff403d',
  interactive: '#3770f6',
}

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof defaultColors
    grays: {
      gray5: string
      gray10: string
      gray20: string
      gray40: string
      gray90: string
    }
    spacing: {
      small: string
      medium: string
      large: string
    }
    padding: {
      small: string
      medium: string
      large: string
    }
    borderRadius: {
      small: string
      medium: string
      large: string
    }
    borderWidth: {
      light: string
      medium: string
      heavy: string
    }
    fontSize: {
      body: string
      h1: string
      h2: string
      h3: string
    }
  }
}

export type BrandColors = typeof defaultColors

export const defaultTheme: DefaultTheme = {
  colors: defaultColors,
  grays: {
    gray5: '#f5f5f7',
    gray10: '#dde1e6',
    gray20: '#c0c9d4',
    gray40: '#656d7f',
    gray90: '#0f1111',
  },
  spacing: {
    small: '6px',
    medium: '10px',
    large: '20px',
  },
  padding: {
    small: '6px',
    medium: '10px',
    large: '20px',
  },
  borderRadius: {
    small: '4px',
    medium: '8px',
    large: '12px',
  },
  borderWidth: {
    light: '0.5px',
    medium: '1px',
    heavy: '2px',
  },
  fontSize: {
    body: '12px',
    h1: '24px',
    h2: '18px',
    h3: '15px',
  },
}
