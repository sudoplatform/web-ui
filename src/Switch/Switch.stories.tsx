import { storiesOf } from '@storybook/react'
import React from 'react'

import { Switch } from './Switch'

storiesOf('Switch', module).add('default', () => (
  <Switch defaultChecked={true} onChange={(value) => alert(value)} />
))
