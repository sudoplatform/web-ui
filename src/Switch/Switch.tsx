import { Switch as AntdSwitch } from 'antd'
import { SwitchProps } from 'antd/lib/switch'
import React from 'react'
import styled from 'styled-components'

const StyledSwitch = styled(AntdSwitch)`
  &.ant-switch-checked {
    background-color: ${({ theme }) => theme.colors.primary};
  }

  .ant-click-animating-node {
    visibility: hidden;
  }
`

export const Switch: React.FC<SwitchProps> = (props) => {
  return <StyledSwitch {...props} />
}
