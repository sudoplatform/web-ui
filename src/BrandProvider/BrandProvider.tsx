import 'antd/dist/antd.min.css'

import React from 'react'
import { ThemeProvider } from 'styled-components'

import { BrandColors, defaultTheme } from './theme'

interface Props {
  colors?: BrandColors
}

export const BrandProvider: React.FC<Props> = (props) => {
  const theme = {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...props.colors,
    },
  }

  return <ThemeProvider theme={theme}>{props.children}</ThemeProvider>
}
