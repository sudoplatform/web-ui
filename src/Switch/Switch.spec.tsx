import React from 'react'
import { create } from 'react-test-renderer'

import { BrandProvider } from '../BrandProvider'
import { Switch } from './Switch'

describe('Switch', () => {
  it('should render', () => {
    const wrapper = create(
      <BrandProvider>
        <Switch />
      </BrandProvider>,
    )
    const serialized = wrapper.toJSON()
    expect(serialized).toMatchSnapshot()
  })
})
