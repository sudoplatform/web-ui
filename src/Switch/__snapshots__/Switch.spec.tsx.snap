// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render 1`] = `
.c0.ant-switch-checked {
  background-color: chocolate;
}

.c0 .ant-click-animating-node {
  visibility: hidden;
}

<button
  aria-checked={false}
  className="ant-switch c0"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="switch"
  type="button"
>
  <div
    className="ant-switch-handle"
  />
  <span
    className="ant-switch-inner"
  />
</button>
`;
